src/aiq/tool/code_execution/local_sandbox/local_sandbox_server.py

```diff
@@ -38,43 +40,98 @@ def execute_python(generated_code, timeout):
     # running in a separate process to ensure any kind of crashes are properly handled
     queue = multiprocessing.Queue()
     process = multiprocessing.Process(target=execute_code_subprocess, args=(generated_code, queue))
+    
     process.start()
     process.join(timeout=timeout)
 
     if process.is_alive():  # didn't finish successfully
         process.kill()
         return {"process_status": "timeout", "stdout": "", "stderr": "Timed out\n"}
 
-    return queue.get()
+    result = queue.get()
+    return result
 
 
 # need to memory-limit to avoid common errors of allocating too much
 # but this has to be done in a subprocess to not crush server itself
 def execute_code_subprocess(generated_code, queue):
+    print(f"[DEBUG] execute_code_subprocess started, PID: {os.getpid()}")
+    
     limit = 1024 * 1024 * 1024 * 10  # 10gb - somehow with a smaller limit the server dies when numpy is used
     resource.setrlimit(resource.RLIMIT_AS, (limit, limit))
     resource.setrlimit(resource.RLIMIT_DATA, (limit, limit))
 
-    # this can be overriden inside generated code, so it's not a guaranteed protection
-    sys.stdout = StringIO()
+    # Capture both stdout and stderr
+    old_stdout = sys.stdout
+    old_stderr = sys.stderr
+    stdout_capture = StringIO()
+    stderr_capture = StringIO()
+    
     try:
+        sys.stdout = stdout_capture
+        sys.stderr = stderr_capture
         exec(generated_code, {})  # pylint: disable=W0122
-        queue.put(sys.stdout.getvalue())
+        
+        stdout_content = stdout_capture.getvalue()
+        stderr_content = stderr_capture.getvalue()
+        
+        print(f"[DEBUG] Code executed successfully")
+        print(f"[DEBUG] stdout length: {len(stdout_content)}")
+        print(f"[DEBUG] stderr length: {len(stderr_content)}")
+        
+        queue.put({
+            "process_status": "completed",
+            "stdout": stdout_content,
+            "stderr": stderr_content
+        })
     except Exception as e:
-        print(f"Error: {str(e)}")
-        queue.put({"process_status": "error", "stdout": "", "stderr": str(e) + "\n"})
+        print(f"[DEBUG] Exception during code execution: {str(e)}")
+        stderr_capture.write(f"Error: {str(e)}")
+        queue.put({
+            "process_status": "error", 
+            "stdout": stdout_capture.getvalue(), 
+            "stderr": stderr_capture.getvalue()
+        })
+    finally:
+        sys.stdout = old_stdout
+        sys.stderr = old_stderr
+        print("[DEBUG] Restored stdout/stderr")
```

Comment:
Should use logging rather than raw print -- please apply to all prints in this source file.